src/executor/graph_executor.cc

```diff
@@ -767,6 +767,28 @@ void GraphExecutor::InitOpSegs() {
   return;
 }
 
+void GraphExecutor::ExecuteMonCallback(size_t nid) {
+  static const auto& flist_outputs =
+      nnvm::Op::GetAttr<nnvm::FListOutputNames>("FListOutputNames");
+  const auto& idx = graph_.indexed_graph();
```

Comment:
https://github.com/dmlc/nnvm/blob/0aa3ff6fd561789b27242f257593962080f2de88/src/core/graph.cc#L12
It's not constructing a new graph every time.